datafusion/sql/tests/cases/diagnostic.rs

```diff
@@ -201,14 +201,8 @@ fn test_ambiguous_reference() -> Result<()> {
     let diag = do_query(query);
     assert_eq!(diag.message, "column 'first_name' is ambiguous");
     assert_eq!(diag.span, Some(spans["a"]));
-    assert_eq!(
-        diag.notes[0].message,
-        "possible reference to 'first_name' in table 'a'"
-    );
-    assert_eq!(
-        diag.notes[1].message,
-        "possible reference to 'first_name' in table 'b'"
-    );
+    assert_eq!(diag.notes[0].message, "possible column a.first_name");
+    assert_eq!(diag.notes[1].message, "possible column b.first_name");
```

Comment:
What would happened if a very similar column (e.g. `first_neme`) was in scope. I guess they would see `"possible column a.first_neme"` even though there wasn't really ambiguity there because the planner didn't match that column.

I don't think this is desirable. What do you think @comphead @alamb

Reply:
My feeling ambiguity might be okay, the final choice is on the user but what is definitely  needs to avoid ig leventshein gets false positives and get 0.5 match. in this case the closest column name will be proposed(which can be totally different from what user typed) but others also hidden. If that happen user will have to do more action like `EXPLAIN` to find the schema. Not sure how to test such scenario though

Reply:
Ok if you say non-exact matches should be shown too, I'm on board.